bsp/raspberrypi/rp2040/src/hal/spi.zig

```diff
@@ -72,39 +156,104 @@ pub const SPI = enum(u1) {
     pub inline fn is_readable(spi: SPI) bool {
         return spi.get_regs().SSPSR.read().RNE == 1;
     }
-    pub fn transceive(spi: SPI, src: []const u8, dst: []u8) usize {
+
+    fn validate_bitwidth(PacketType: type) void {
+        const acceptable_types = .{ u4, u5, u6, u7, u8, u9, u10, u11, u12, u13, u14, u15, u16 };
+        for (acceptable_types) |acceptable_type| {
+            if (std.meta.eql(PacketType, acceptable_type)) return;
+        }
+        @compileError("PacketType must be one of u4, u5, ..., u16");
+    }
+
+    const fifo_depth = 8;
+
+    /// Disable SPI, pre-fill the TX FIFO as much as possible, and then re-enable to start transmission.
+    /// Leads to performance gains in thoroughput. Returns how many bytes were consumed from src.
+    fn prime_tx_fifo(spi: SPI, PacketType: type, src: []const PacketType) usize {
```

Comment:
`PacketType` should be comptime

Reply:
General question, I usually omit `comptime` on a `type` arg because they _have_ to be `comptime` given they're types. I've seen this a decent amount in other Zig code. No issue adding `comptime` to be explicit though!

Reply:
I would add them just to make clear that this parameter is comptime only. It was necessary for a long time, so it might not be "best practise" nowadays, but i still like to see quickly if something has comptime effects